### DISCUSSION TITLE
Configuration to discourage or remove the multi-package flow

### DISCUSSION THREAD

**Discussion by Rugvip** [changesets/changesets#1018] [Ideas]
Over in [backstage](https://github.com/backstage/backstage) we're happy users of changesets, and have been using it for many years. Something that we've been struggling with quite a lot is the external contributions and how write good changesets. A particular issue that pops up very often are changesets that are written towards multiple packages at once, leading to a confusing changelog. For example:

```yaml
---
'@acme/components': patch
'@acme/theme': patch
---

Fixed an issue where the `Button` component didn't animate correctly.

Tweaked background contrasts.
```

Reading the changeset as is the information makes sense, but once you split it out into the changelogs for the `components` and `theme` packages, there's a bunch of unrelated information. The `components` package doesn't decide the background colors, and the `theme` package doesn't have a `Button` component.

I've been thinking that it's worth tweaking the CLI UX to either outright disallow selection of multiple packages at once, or in some way heavily discourage it. That way I'm thinking you're more likely to be writing changesets for one packages at a time, but you still might have the convenience of applying it to multiple packages in the cases where that makes sense.

**Comment by Andarist** [comment]
Happy to brainstorm this idea. I think that the laid-out problem is real - people are often using Changesets "incorrectly" (I'm stretching it a little bit as there is no single way to use Changesets). The fact that you can create multiple changesets per PR is an explicit feature. It's very much by design. However, it doesn't always "click" for people. 

I'm not entirely sure how to achieve a great UX here though as this all boils down to some details. If we just communicate this in the CLI prompts then... nobody will actually read it :sweat_smile: 

Perhaps a config flag that would disallow this (require a single package per changeset) + a CLI flag that would bypass this config-level check when creating prompts would do the trick?
